model_analyzer/config/config_union.py

```diff
@@ -58,13 +60,44 @@ def set_value(self, value):
             The value for this field.
         """
 
+        config_statuses = []
         for i, type_ in enumerate(self._types):
-            status = type_.set_value(value)
-            if status == MODEL_ANALYZER_SUCCESS:
+            config_status = type_.set_value(value)
+            config_statuses.append(config_status)
+            if config_status.status() == CONFIG_PARSER_SUCCESS:
                 self._used_type_index = i
                 return super().set_value(type_)
         else:
-            return MODEL_ANALYZER_FAILURE
+            message = f'Value "{value}" cannot be set for field "{self.name()}".'
+            message += ' This field allows multiple types of values.'
+            message += ' You only need to fix one of the errors below:\n'
```

Comment:
Here you can also do
```
message = (
    f'Value "{value}" cannot be set for field "{self.name()}".'
    ' This field allows multiple types of values.'
    ' You only need to fix one of the errors below:\n'
)
```

I'm not 100% sure but I think this will be less copies. If the above doesn't work maybe the single quotes and double quotes need to be exchanged.